sdk/core/core/inc/az_span.h

```diff
@@ -15,51 +15,30 @@
 
 #include <_az_cfg_prefix.h>
 
-/**
- * An immutable span of bytes (octets).
- */
-typedef struct {
-  /**
-   * Points to the first byte.
-   */
-  uint8_t const * begin;
-  /**
-   * A number of bytes in the span.
-   */
-  size_t size;
-} az_const_span;
+#define AZ_SPAN_TYPE(NAME, VALUE) \
+  typedef struct { \
+    VALUE * begin; \
+    size_t size; \
+  } NAME;
 
 /**
- * A mutable span of bytes.
+ * An immutable span of bytes (octets).
  */
-typedef struct {
-  uint8_t * begin;
-  size_t size;
-} az_span;
+AZ_SPAN_TYPE(az_span, uint8_t const)
 
 typedef az_result az_result_byte;
 
-AZ_INLINE bool az_span_is_empty(az_span const span) { return !(span.size > 0); }
-
-AZ_INLINE bool az_const_span_is_empty(az_const_span const span) { return !(span.size > 0); }
+AZ_INLINE bool az_span_is_empty(az_span const span) { return span.size <= 0; }
 
 AZ_INLINE bool az_span_is_valid(az_span const span) {
-  return span.begin == NULL ? span.size == 0
-                            : ((span.size == 0 || span.size > 0)
-                               && (span.size == 0 || span.begin <= (span.begin + span.size - 1)));
-}
-
-AZ_INLINE bool az_const_span_is_valid(az_const_span const span) {
-  return span.begin == NULL ? span.size == 0
-                            : ((span.size == 0 || span.size > 0)
-                               && (span.size == 0 || span.begin <= (span.begin + span.size - 1)));
+  return span.size == 0 || (span.begin != NULL && span.begin <= span.begin + span.size - 1);
 }
 
 /**
  * Returns a byte in `index` position.
  * Returns `AZ_ERROR_EOF` if the `index` is out of the span range.
```

Comment:
formatting here, although I think the returns annotation got mangled someplace, so the design guide needs updating there